Example/Database/Tests/Integration/FIRDatabaseTests.m

```diff
@@ -447,7 +447,13 @@ - (FIRDatabase *) databaseForURL:(NSString *)url {
 }
 
 - (FIRDatabase *) databaseForURL:(NSString *)url name:(NSString *)name {
-    id app = [[FIRFakeApp alloc] initWithName:name URL:url];
+    id app = nil;
+    NSString *defaultDatabaseURL = [NSString stringWithFormat:@"url:%@", self.databaseURL];
+    if ([url isEqualToString:self.databaseURL] && [name isEqualToString:defaultDatabaseURL]) {
+        app = [FIRApp defaultApp];
```

Comment:
What's the reason for using the default app here?

Reply:
The database that's returned here needs to be the same between calls, otherwise no one holds on to the app that's created here and a new fake app gets called between `[self defaultDatabase]` calls.

Since the tests that use this check that the database instances are correct for the same app, and there isn't a list of databases keyed by app, I just used the default app since this is in the integration tests and the default app should be valid.

Is there a simpler way to achieve those tests that I missed?

Reply:
Discussed offline, comment added. Thanks!